NanamiL3.java

```diff
@@ -0,0 +1,122 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+// import packages to read user input from console
+
+public class NanamiL3 {
+    private static Task[] tasklist = new Task[100];
+    private static int taskcount;
+    private int taskCount = 0;
+    private static boolean applicationOpen = true;
+
+    public static void main(String args[]) {
+        // driver code to begin the application
+        giveIntroduction();
+
+        while (applicationOpen) {
+            String input = readUserInput();
+            String [] commandWords = input.split("\\s+"); // cuts phrases at spaces into an array
+
+            // perform actions based on user input
+            if (input.equals("bye") || input.equals("Bye")) {
+                // if the command is exactly 'bye' or 'Bye'
+                giveFarewell();
+                applicationOpen = false;
+            }
+            else if (input.equals("list") || input.equals("List")) {
+                // if the command is exactly 'list' or 'List'
+                displayTaskList();
+            }
+            else if (commandWords[0].equals("mark") || commandWords[0].equals("Mark")) {
+                int taskNumber = checkIfNumber(commandWords[1]);
+                // if the first word is mark and the second word is a number
+                if (taskNumber < taskcount) { // if the task exists in the tasklist
+                    tasklist[taskNumber].markAsDone();
+                    System.out.println("I've marked [" + taskNumber + "] as completed.");
+                }
+                else {
+                    System.out.println("» This task number does not exist.");
+                }
+                displayTaskList();
+            }
+            else if (commandWords[0].equals("unmark") || commandWords[0].equals("Unmark")) {
+                int taskNumber = checkIfNumber(commandWords[1]);
+                // if the first word is unmark and the second word is a number
+                if (taskNumber < taskcount) { // if the task exists in the tasklist
+                    tasklist[taskNumber].markAsUndone();
+                    System.out.println("I've marked [" + taskNumber + "] as incompleted.");
+                }
+                else {
+                    System.out.println("» This task number does not exist.");
+                }
+                displayTaskList();
+            }
+            else {
+                addToList(input);
+            }
+        }
+    }
```

Comment:
Do consider avoiding long methods and consider taking corrective action when it goes beyond 30 LoC.